.gitlab/kernel_version_testing/system_probe.yml

```diff
@@ -404,16 +407,30 @@ kernel_matrix_testing_cleanup:
   script:
     - |
       if [ -f $CI_PROJECT_DIR/stack.outputs ]; then
-        INSTANCE_IPS=$(cat $CI_PROJECT_DIR/stack.outputs | grep -e '-instance-ip' | cut -d ' ' -f 2 | tr '\n' ' ' | awk '{$1=$1};1' | tr ' ' ',')
+        INSTANCE_IPS=$(cat $CI_PROJECT_DIR/stack.outputs | grep -e "$ARCH-instance-ip" | cut -d ' ' -f 2 | tr '\n' ' ' | awk '{$1=$1};1' | tr ' ' ',')
         IP_FILTER="Name=private-ip-address,Values=$INSTANCE_IPS"
       else
         echo "No ${CI_PROJECT_DIR}/stack.outputs file found"
         IP_FILTER=""
```

Comment:
If no `stack.outputs` file is found, will this cause it to terminate a wider selection of instances? Should we exit here?

Reply:
Since we are filtering by pipeline id, it should only terminate instances launched as part of this pipeline.

Moreover, we want the cleanup job to run even when the setup job failed, and no `stack.outputs` file is present so that leftover instances are terminated.
As far as I know the `IP_FILTER` is redundant because the pipeline-id is always unique. We can consider removing this filter altogether.

Reply:
cleanup is per-arch now, so we do need the IP_FILTER

Reply:
That was more my point, in that the failing of a single setup job would cause the termination of both metal instances.